app/models/amusement_park.rb

```diff
@@ -1,3 +1,13 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanic_rides, through: :rides
+  has_many :mechanics, through: :mechanic_rides
+
+  def hired_mechanics
+    mechanics.distinct
+  end
+
+  # def rides_by_avg_experience
+  #   I was trying to join merchants to @six_flags.rides, average(merchants.years_experience) AS avg_mech_experience, and order by avg_mech_experience but couldn't get it working in time
```

Comment:
No prob! If you tackle it later, make sure you add a group by in there to group by rides

Reply:
If you want to see a sample query, look below (ideally AFTER you've tried this extension on your own!)
.
.
.
.
.
.
.
.
.
.
.
```
    rides.joins(:mechanics)
         .select("rides.name, AVG(mechanics.years_experience) AS avg_mech_exp")
         .group("rides.id")
         .order("avg_mech_exp DESC")
```